proposals/0220-support-for-android-custom-routerservice.md

```diff
@@ -43,10 +56,21 @@ Currently SdlProxy identifies the RouterService in three steps:
 The problem is that custom RouterService is unlikely chosen in step #2.
 Step #1 actually depends on the timing of when the app calls queryForConnectedService. It is up to the application when queryForConnectedService is called, but [the integration-basic document](https://smartdevicelink.com/en/guides/android/getting-started/integration-basics/) suggests it should be called at Main Activities onCreate.
 
+#### The cases where this proposal focuses on
+The following section focuses on the case other than "a trigger sent from RouterService". For example, in the following cases, the app will start proxy:
+- When OnLanguageChange is notified: in this case, the app has to restart the proxy.
+- When a user removes the app from Recent List: in this case, ActvityManager may automatically restarts the killed service, and restarts proxy in there. RouterService internally uses DeathRecipient interface to detect binderDied case, which might help some cases, but not all the cases.
+- When a user enforces the app to stop by settings | Apps | force stop: in this case ActivityManager may automatically restarts the killed service as well as the remove from Recent list case.
+- An app may pay attention to some error case (e.g. onProxyClosed, etc), and then restarts the proxy to get recovered.
+- An app may restart proxy when yet another transport is connected. This might be useful especially if the SDL core does not support secondary transport. For instance, if BT transport is connected, the app uses NON_MEDIA layout. When user connected USB or WIFI, then the app restarts the proxy instance and then uses NAV_FULLSCREEN_MAP layout.
```

Comment:
```suggestion
- An app may restart proxy when yet another transport is connected. This might be useful especially if the SDL core does not support secondary transport. For instance, if BT transport is connected, the app uses NON_MEDIA layout. When user connects USB or WIFI, then the app restarts the proxy instance and then uses NAV_FULLSCREEN_MAP layout.
```